packages/flutter/lib/src/widgets/page_storage.dart

```diff
@@ -129,7 +113,123 @@ class PageStorageBucket {
   }
 }
 
-/// A widget that establishes a page storage bucket for this widget subtree.
+/// Establish a subtree in which widgets can opt into persisting states after
+/// being destroyed.
+///
+/// [PageStorage] is used to save and restore values that can outlive the widget.
+/// For example, when multiple pages are grouped in tabs, when a page is
+/// switched out, its widget is destroyed and its state is lost. By adding a
+/// [PageStorage] at the root and adding a [PageStorageKey] to each page, few
+/// page's state, like Scrollable Widgets, will be stored automatically in its closest ancestor [PageStorage], and
+/// restored when it's switched back. Only the state knowing how to use [PageStorage] is persisted.
```

Comment:
What do you mean by "Only the state knowing how to use [PageStorage] is persisted."? Maybe remove this sentence altogether?